src/helpers.rs

```diff
@@ -69,6 +75,54 @@ fn try_resolve_did<'tcx>(tcx: TyCtxt<'tcx>, path: &[&str]) -> Option<DefId> {
     )
 }
 
+/// if the primitive provided can be converted to a type matching the hir type pattern
+/// prim_type_pat (conversion through prim_conv_fct), then create a CArg of this primitive
+/// value with the carg_constructor provided
```

Comment:
except for the `return` part, all of this logic could be regular Rust functions. I think I'd prefer that, and then calling the `return` manually.

The `prim_type_pat` can just be matched by equality instead of destructuring for this to work.